Arelle/Arelle
# Update pyparsing

#### Reason for change
The committed version of pyparsing (1.5.5) is very old (Aug 12, 2010).

#### Description of change
Replaces the committed version of pyparsing with the pip installed version.

#### Steps to Test
* XPath parser
 * CI/conformance suites (formula)
* Sphinx plugin
  * From the GUI enable the Sphinx plugin.
  * Load a US GAAP filing.
  * From the file menu "Import Sphinx files...".
  * Select `arelle/plugin/sphinx/US-GAAP Ratios Example.xsr`.
  * Validate and confirm no exceptions are thrown.
* Review the [pyparsing_py3.py diff](https://github.com/Arelle/Arelle/files/9941160/pyparsing_py3.py.txt) between the committed version of `pyparsing_py3.py` and upstream (v1.5.5).

**review**:
@Arelle/arelle
